apps/studio/components/interfaces/APIKeys/APIKeyDeleteDialog.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react'
+import { toast } from 'sonner'
+import { useParams } from 'common/hooks'
+import { useAPIKeyDeleteMutation } from 'data/api-keys/api-key-delete-mutation'
+import TextConfirmModal from 'ui-patterns/Dialogs/TextConfirmModal'
+import { APIKeysData } from 'data/api-keys/api-keys-query'
+import { DropdownMenuItem } from 'ui'
+
+const APIKeyDeleteDialog = ({
+  apiKey,
+}: {
+  apiKey: Extract<APIKeysData[number], { type: 'secret' | 'publishable' }>
+}) => {
+  const { ref: projectRef } = useParams()
+  const [isOpen, setIsOpen] = useState(false)
+
+  const { mutate: deleteAPIKey, isLoading: isDeletingAPIKey } = useAPIKeyDeleteMutation({
+    onSuccess: () => {
+      toast.success(`Successfully deleted API key`)
+      setIsOpen(false)
+    },
+  })
+
+  const onDeleteAPIKey = () => {
+    if (!projectRef) return console.error('Project ref is required')
+    deleteAPIKey({ projectRef, id: apiKey.id })
+  }
+
+  const canDeleteAPIKeys = true // todo
```

Comment:
Need to check for proper permissions here

Reply:
can refer from API to properly hook up

![image](https://github.com/user-attachments/assets/26de824a-9b80-4a6d-87b4-6732b4f422c3)

Reply:
pushed up the fix